File: Python/BOJ/2635 수 이어가기.py
import sys
N = int(sys.stdin.readline())
result = []

for i in range(1, N+1):
    now, next = N, i
    temp = [N, i]
    while now >= next:
        x = now - next
        temp.append(x)
        now, next = next, x
        if len(temp) > len(result):
            result = temp

print(len(result))
print(*result)

File: Java/BOJ/2635 수 이어가기.java
import java.io.*;
import java.util.*;
import java.math.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        List<Integer> result = new ArrayList<>();
        for (int i = 1 ; i < N+1; i++) {
        	int next = i;
        	int now = N;
        	List<Integer> temp = new ArrayList<>();
        	temp.add(now);
        	temp.add(i);
        	while(now >= next) {
        		int x = now - next;
        		temp.add(x);
        		now = next;
        		next = x;
        		if (temp.size() > result.size()) {
        			result = temp;
        		}
        	}
        }
        System.out.println(result.size());
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < result.size(); i++) {
        	sb.append(result.get(i)).append(" ");
        }
        System.out.println(sb);
    }
}